contractcourt/chain_watcher.go

```diff
@@ -1182,6 +1169,19 @@ func (c *chainWatcher) dispatchContractBreach(spendEvent *chainntnfs.SpendDetail
 	log.Infof("Breached channel=%v marked pending-closed",
 		c.cfg.chanState.FundingOutpoint)
 
+	// With the event processed and channel closed, we'll now notify all
+	// subscribers of the event.
+	c.Lock()
+	for _, sub := range c.clientSubscriptions {
```

Comment:
Excellent catch! 

I think this might be the culprit of some of the flakes we've seen in this area as well.